src/moa/moa_controllers/moa_controllers/ack_to_can.py

```diff
@@ -0,0 +1,119 @@
+#/usr/bin/env python3
+from typing import Optional
+import rclpy
+from rclpy.node import Node
+import time
+
+from ackermann_msgs.msg import AckermannDriveStamped
+from ackermann_msgs.msg import AckermannDrive
+from moa_msgs.msg import CANStamped
+from std_msgs.msg import Header
+from builtin_interfaces.msg import Time
+
+import numpy as np
+import zlib
+from math import pi
+
+
+def compress_floats(values:np.ndarray) -> bytes:    
+    # Compute the differences between consecutive values
+    differences = np.diff(values)
+
+    # Compress the differences using zlib
+    compressed = zlib.compress(differences, level=9)
+
+    return compressed
+
+
+def ackermann_to_can_parser(node: Node, ack_msg: AckermannDriveStamped) -> Optional[CANStamped]: #type for option optionalnull
+        #checks before sending Ackermann
+        if 0 < ack_msg.drive.speed > 120/3.6: #m/s   
+            node.get_logger().warn(f'ackermann drive SPEED out of bounds')
+            return None
+        
+        elif 0 < ack_msg.drive.acceleration > 256: #m/s^2
+            node.get_logger().warn(f'ackermann drive ACCLERATION out of bounds')
+            return None
+
+        elif 0 < ack_msg.drive.jerk: #m/s^3 
+            #not too fussed about assign 1 byte (minifloat)
+            node.get_logger().warn(f'ackermann drive JERK out of bounds')
+            return None
+        
+        elif -45*pi/180 < ack_msg.drive.steering_angle > 45*pi/180: #radians
+            node.get_logger().warn(f'ackermann drive STEERING_ANGLE out of bounds')
+            return None
+        
+        elif 0 < ack_msg.drive.steering_angle_velocity < 1: #radians/s
+            #unsure of upper limit def dont need more than 1
+            node.get_logger().warn(f'ackermann drive STEERING_ANGLE_VELOCITY out of bounds')
+            return None
+        
+        #format values of Ackermann
+        ackermann_vals = [
+            np.uint8(round(ack_msg.drive.speed)), #(functionally limited to 100-120km/h) integer and round 
+            np.uint8(round(ack_msg.drive.acceleration)), #need to check between 0-256 
+            np.uint8(ack_msg.drive.jerk*1000), #need to check between 0-256 if dont need document that dont need and don't parse
+            np.float16(ack_msg.drive.steering_angle), #can be negative (assume between +-45 degrees change to radians) make float16
+            np.uint8(ack_msg.drive.steering_angle_velocity*1000), #can be negative between 0-1 (take 1 byte 8 bit minifloat)
+            ]
+
```

Comment:
There is no need for compression if the data is just this bc its 6 bytes and we have 8 to use and the compression will just take up compute resourses